src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/PropertyTabAttribute.cs

```diff
@@ -145,34 +145,25 @@ private void InitializeTabClasses()
         public PropertyTabScope[] TabScopes { get; private set; }
 
         public override bool Equals([NotNullWhen(true)] object? other)
-        {
-            if (other is PropertyTabAttribute propertyTabAttribute)
-            {
-                return Equals(propertyTabAttribute);
-            }
-            return false;
-        }
+            => Equals(other as PropertyTabAttribute);
 
-        public bool Equals(PropertyTabAttribute other)
+        public bool Equals([NotNullWhen(true)] PropertyTabAttribute? other)
         {
-            if (other == (object)this)
-            {
+            if (other is null)
+                return false;
+
```

Comment:
NIT: Isn't this null check redundant? `ReferenceEquals` allows nulls
```suggestion
```

Reply:
But if ReferenceEquals returns false we'll have NRE in the next if
I can swap null-check and ReferenceEquals, but does it really matter?